lib/src/layout_element.dart

```diff
@@ -251,11 +251,74 @@ TableStyleElement parseTableDefinitionElement(
   }
 }
 
+class DetailsContentElement extends LayoutElement {
+  List<dom.Element> elementList;
+
+  DetailsContentElement({
+    String name,
+    List<StyledElement> children,
+    dom.Element node,
+    this.elementList,
+  }) : super(name: name, node: node, children: children);
+
+  @override
+  Widget toWidget(RenderContext context) {
+    return ExpansionTile(
+        title: children != null && elementList.first.localName == "summary" ? StyledText(
+          textSpan: TextSpan(
+            style: style.generateTextStyle(),
+            children: [children
+                .map((tree) => context.parser.parseTree(context, tree))
+                .toList().first] ??
+                [],
+          ),
+          style: style,
+        ) : Text("Details"),
+        children: [
+          StyledText(
+            textSpan: TextSpan(
+              style: style.generateTextStyle(),
+              children: getChildren(children, context)
+            ),
+            style: style,
+          ),
+        ]
+    );
+  }
+
+  List<InlineSpan> getChildren(List<StyledElement> children, RenderContext context) {
+    if (children.map((tree) => context.parser.parseTree(context, tree)).toList() == null) {
+      return [];
+    } else {
+      List<InlineSpan> reducedChildren = children.map((tree) => context.parser.parseTree(context, tree)).toList();
+      reducedChildren.removeAt(0);
+      return reducedChildren;
+    }
+  }
+}
+
+class EmptyLayoutElement extends LayoutElement {
+  EmptyLayoutElement({String name = "empty"}) : super(name: name);
+
+  @override
+  Widget toWidget(_) => null;
+}
+
 LayoutElement parseLayoutElement(
-  dom.Element element,
-  List<StyledElement> children,
+    dom.Element element,
+    List<StyledElement> children,
 ) {
   switch (element.localName) {
+    case "details":
+      if (children == null || children.isEmpty) {
```

Comment:
This is better written `if (children?.isEmpty ?? false)`

Reply:
Yeah that is nicer, I'll update that. Learning new things every day.... thanks!